x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/storage/ReactiveStorageDeciderService.java

```diff
@@ -118,7 +124,20 @@ public AutoscalingDeciderResult scale(Settings configuration, AutoscalingDecider
             .total(autoscalingCapacity.total().storage().getBytes() + unassignedBytes + assignedBytes, null)
             .node(maxShardSize, null)
             .build();
-        return new AutoscalingDeciderResult(requiredCapacity, new ReactiveReason(message, unassignedBytes, assignedBytes));
+        return new AutoscalingDeciderResult(
+            requiredCapacity,
+            new ReactiveReason(
+                message,
+                unassignedBytes,
+                assignedBytes,
+                unassignedBytesUnassignedShards.shardIds(),
```

Comment:
I think we mentioned that it would be nice to provide some the allocation debug info for few of the unassigned shards, so it is easier to understand why a shard is unassigned? wdyt @henningandersen?

Reply:
We discussed that [here](https://github.com/elastic/elasticsearch/pull/86132#discussion_r866652526) and I suggested to keep that to a separate PR. Do you think it need to be incorporated into this PR?

Reply:
Oh sorry, I missed that. I think it's fine to tackle that in a separate PR.